cms/djangoapps/import_from_modulestore/views/v0/views.py

```diff
@@ -1,317 +1,81 @@
-from opaque_keys.edx.keys import UsageKey
 from opaque_keys.edx.locator import LibraryLocatorV2
+from openedx_learning.api.authoring_models import DraftChangeLog
 from rest_framework import status
-
 from rest_framework.permissions import IsAdminUser
-from rest_framework.generics import CreateAPIView, RetrieveAPIView
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
-from cms.djangoapps.import_from_modulestore import api
+from cms.djangoapps.import_from_modulestore.api import import_course_to_library
 from cms.djangoapps.import_from_modulestore.models import Import
 from cms.djangoapps.import_from_modulestore.permissions import IsImportAuthor
-from cms.djangoapps.import_from_modulestore.views.v0.serializers import ImportBlocksSerializer, ImportSerializer
+from cms.djangoapps.import_from_modulestore.views.v0.serializers import (
+    CreateImportTaskSerializer,
+    ImportTaskStatusSerializer,
+)
 from openedx.core.djangoapps.content_libraries.api import ContentLibrary
 from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
 
 
-class ImportBlocksView(APIView):
+class ImportView(APIView):
     """
-    Import blocks from a course to a library.
-
-    **Example Request**
-        POST /api/import_from_modulestore/v0/import_blocks/
-
-    **Example request data**
-        ```
-        {
-            "usage_ids": ["block-v1:org+course+run+type@problem+block@12345"],
-            "target_library": "lib:org:test",
-            "import_uuid": "78df3b2c-4e5a-4d6b-8c7e-1f2a3b4c5d6e",
-            "composition_level": "xblock",
-            "override": false
-        }
-        ```
-
-    **POST Parameters**
-        - usage_ids (list): A list of usage IDs of the blocks to be imported.
-        - target_library (str): The library to which the blocks will be imported.
-        - import_uuid (str): The UUID of the import task.
-        - composition_level (str): The composition level of the blocks to be imported.
-        - override (bool): Whether to override existing blocks in the library.
-
-    **Responses**
-        - 200: Import blocks from a course to a library task successfully started.
-        - 400: Invalid request data.
-        - 401: Unauthorized.
-        - 403: Forbidden, request user is not the author of the received import.
-        - 404: Import not found.
-
-    **Example Response**:
-        ```
-        {
-            "status": "success"
-        }
-        ```
+    Import course content from modulestore into a content library.
     """
 
-    serializer_class = ImportBlocksSerializer
-
-    permission_classes = (IsImportAuthor,)
+    permission_classes = (IsAdminUser, IsImportAuthor)
     authentication_classes = (
         JwtAuthentication,
         BearerAuthenticationAllowInactiveUser,
         SessionAuthenticationAllowInactiveUser,
     )
+    serializer_class = ImportTaskStatusSerializer
+    method_serializers = {
```

Comment:
I think that after adding the above-mentioned fields to the model, we can try to get by with one serializer.